pkg/virt-controller/watch/snapshot/snapshot_test.go

```diff
@@ -1912,7 +1912,11 @@ func expectVMSnapshotContentCreate(client *kubevirtfake.Clientset, content *snap
 		Expect(ok).To(BeTrue())
 
 		createObj := create.GetObject().(*snapshotv1.VirtualMachineSnapshotContent)
-		Expect(createObj).To(Equal(content))
+		Expect(createObj.ObjectMeta).To(Equal(content.ObjectMeta))
+		Expect(createObj.Spec.Source).To(Equal(content.Spec.Source))
+		Expect(createObj.Spec.VirtualMachineSnapshotName).To(Equal(content.Spec.VirtualMachineSnapshotName))
+		Expect(createObj.Spec.VolumeBackups).To(Equal(content.Spec.VolumeBackups))
+		Expect(createObj.Status).To(Equal(content.Status))
```

Comment:
On the one hand, this will make the error much more compact and readable.
On the other, the test quality itself is being hurt a bit, since we now don't check full equality. In the future, when more important fields would be added, one of them might be missed in testing.

I've tried to find ways to adjust the output in gomega's documentation[1][2], but couldn't find something that's good enough. The options are either to change a global variable (yuck), or to implement ourselves the conversion to a string representation.

Therefore, let me suggest the following idea:
Keep the checks as they are right now, and in the end add the now deleted `Expect(createObj).To(Equal(content))` check. This way, if one of the above fields is not as expected, we would fail with a nice and compact error. If in the future we forget to check some field, we will fail on `Expect(createObj).To(Equal(content))` with an ugly message, but at least we would know we missed something.

[1] - https://onsi.github.io/gomega/#adjusting-output
[2] - https://pkg.go.dev/github.com/onsi/gomega/format#GomegaStringer

Reply:
What is the goal of the test and what is really important for us? Do we care for all the fields?

Reply:
This is part of `expectVMSnapshotContentCreate()` function which gets a snapshot content as an argument. So I guess the caller wants the created content to be equal to the one passed as a parameter. My concern is that the current's PR implementation demands manually adding an expectation here if more fields are added, which sounds error-prone to me.

Also bear in mind that the purpose of the PR is not to change the test logic, but just to make the error message better. With what I suggested we achieve this goal without changing any logic.

Reply:
I agree with you @iholder-redhat and I will take your suggestion. Anyways I managed to reproduce the original issue locally and with split it is clearer what is the diff and why. same as for the other test that failed in this PR - suddenly missing volumesnapshotclass. working on it and will update the PR with the fix for that also.